### PR TITLE
Verbose mode and potential fixes.

### PR DESCRIPTION
**Pull request by terjekv**

### CODE REVIEW

File: dynamic/eessi-dynamic [hunk 1]
```diff
@@ -55,6 +57,16 @@ def error(string, exit_code=1):
...
+def ensure_terraform_ssh_keys():
+    """
+    Ensure that ~/.ssh/id_rsa.terraform[.pub] exists.
+    """
+    filepath = "~/.ssh/id_rsa.terraform"
+    fullfilepath = os.path.expanduser(filepath)
+
+    if not os.path.exists(fullfilepath):
+        execute("ssh-keygen -t rsa  -f ~/.ssh/id_rsa.terraform -b 4096 -n ''")
+
```
Comment (boegel) [inline dynamic/eessi-dynamic:69]: please use 2 empty lines between function definitions

File: dynamic/eessi-dynamic [hunk 2]
```diff
@@ -243,30 +263,9 @@ def create_node(node, dryrun=False):
...
-    #         "instance_tag_owner": "terjekv",
-    #         "instance_tag_name": "[terjekv] testnode",
-    #         "instance_tag_user": "terjekv",
-    #         "instance_tag_slack": "@terjekv",
-    #         },
-    #     ],
-    # }
+    nodefile = TERRAFORM_DIRECTORY + "/" + node['name'].lower() + ".tf"
+    create_file_from_template("node.j2", nodefile, node)
+    return nodefile
```
Comment (boegel) [inline dynamic/eessi-dynamic:269]: please use 2 empty lines between function definitions

File: dynamic/eessi-dynamic [hunk 3]
```diff
@@ -360,7 +363,7 @@ def get_hostfile():
             if os.access(osfile, os.W_OK):
                 return osfile
 
-    return '/tmp/hosts'
+    return osfile
```
Comment (boegel) [inline dynamic/eessi-dynamic:367]: please use 2 empty lines between function definitions

**Review by boegel — CHANGES_REQUESTED** [review]

File: dynamic/eessi-dynamic [hunk 4]
```diff
@@ -340,10 +343,24 @@ def create_file_from_template(templatefile, targetfile, data):
         fh.write(outputText)
 
 
+def execute(cmd,print_stdout=False):
```
Comment (boegel) [inline dynamic/eessi-dynamic:346]: Why is `execute` there twice now? Doesn't make much sense to me 😅

File: dynamic/eessi-dynamic [hunk 5]
```diff
@@ -483,4 +502,6 @@ def is_valid_hostname(hostname):
     return all(allowed.match(x) for x in hostname.split("."))
 
 
+main
```
Comment (boegel) [inline dynamic/eessi-dynamic:505]: This looks wrong to me...

**Review by boegel — CHANGES_REQUESTED** [review]

File: dynamic/eessi-dynamic (hunk 5 above)
Comment (terjekv) [reply dynamic/eessi-dynamic]: Whow. Fixed.

File: dynamic/eessi-dynamic (hunk 4 above)
Comment (terjekv) [reply dynamic/eessi-dynamic]: Er, I fail miserably at conflict resolution when blocks are collapsed in VSC. That was neat. Fixed.

**Review by boegel — APPROVED** [review]

**Pull request merged**